src/eva/vision/models/networks/encoders/encoder.py

```diff
@@ -0,0 +1,23 @@
+"""Encoder base class."""
+
+import abc
+from typing import List
+
+import torch
+from torch import nn
+
+
+class Encoder(nn.Module, abc.ABC):
```

Comment:
What is the main motivation for introducing the new encoder / decoder abstractions at this point?
Previously we had backbone / head - how do these play together?

Reply:
in out case backbone outputs a single tensor which encoder a list of them (and the head / decoder process the inputs accordingly)

maybe we could have a conversation about these conventions